bib/internal/progress/progress_test.go

```diff
@@ -0,0 +1,106 @@
+package progress_test
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/osbuild/bootc-image-builder/bib/internal/progress"
+)
+
+func TestProgressNew(t *testing.T) {
+	restore := progress.MockIsattyIsTerminal(true)
+	defer restore()
+
+	for _, tc := range []struct {
+		typ         string
+		expected    interface{}
+		expectedErr string
+	}{
+		{"term", &progress.TermProgressBar{}, ""},
+		{"debug", &progress.DebugProgressBar{}, ""},
+		{"plain", &progress.PlainProgressBar{}, ""},
+		{"bad", nil, `unknown progress type: "bad"`},
+	} {
+		pb, err := progress.New(tc.typ)
+		if tc.expectedErr == "" {
+			assert.NoError(t, err)
+			assert.Equal(t, reflect.TypeOf(pb), reflect.TypeOf(tc.expected), fmt.Sprintf("[%v] %T not the expected %T", tc.typ, pb, tc.expected))
+		} else {
+			assert.EqualError(t, err, tc.expectedErr)
+		}
+	}
+}
+
+func TestPlainProgress(t *testing.T) {
+	var buf bytes.Buffer
+	restore := progress.MockOsStderr(&buf)
+	defer restore()
+
+	// plain progress never generates output
```

Comment:
I suggest moving this comment 2 lines lower, so it's clear that it talks about `SetProgress`